### PR TITLE
Feat/frontend unit tests

### PR DESCRIPTION
**Pull request by nss10**
Jira Ticket: [BRH-353](https://ctds-planx.atlassian.net/browse/BRH-353)

### New Features
* Added unit tests to all the existing ui components.

[BRH-353]: https://ctds-planx.atlassian.net/browse/BRH-353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bmh_admin_portal_ui/src/components/request-workspace/StridesCredits.test.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+    //ensure request Workspace is called when the checkValidity returns true
+    let custom_event_object = {
+        currentTarget: {
+            checkValidity: () => false
+        },
+        preventDefault: () => null
+    };
+    submitFunc(custom_event_object);
+    // mockFunction.mockImplementation(()=>{});
```
Comment (mfshao) [inline bmh_admin_portal_ui/src/components/request-workspace/StridesCredits.test.js:283]: not needed code?

File: bmh_admin_portal_ui/src/views/RequestWorkspace.test.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    waitFor
+} from '@testing-library/react';
+import RequestWorkspace from './request-workspace';
+import * as authUtils from '../util/auth';
+import {
+    act
+} from 'react-dom/test-utils';
+
+
+process.env.REACT_APP_OIDC_AUTH_URI = "https://fence.planx-pla.net/user/oauth2/authorize"
```
Comment (mfshao) [inline bmh_admin_portal_ui/src/views/RequestWorkspace.test.js:14]: why we need to set this?

File: bmh_admin_portal_ui/src/views/WorkspaceAccountsAdmin.test.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+import WorkspaceAccountsAdmin from './workspace-accounts-admin';
+import * as apiUtils from '../util/api';
+import * as authUtils from '../util/auth';
+import { BrowserRouter } from 'react-router-dom';
+
+const tableData = [{
+        "scientific_poc": "Random Sample",
+        "request_status": "active",
+        "total-usage": 10,
+        "account_id": "9032657199",
```
Comment (mfshao) [inline bmh_admin_portal_ui/src/views/WorkspaceAccountsAdmin.test.js:12]: make sure we are not using a real AWS acct ID or root email in here?

**Review by mfshao — APPROVED** [review]
overall I think these unit tests make sense. I hope we can also setup some sort of GH actions along with this PR so these tests can get triggered for PRs so they can serve as branch protections

File: bmh_admin_portal_ui/src/views/RequestWorkspace.test.js (hunk 2 above)
Comment (nss10) [reply bmh_admin_portal_ui/src/views/RequestWorkspace.test.js:14]: Oh yes, I won't be needing this for the `RequestWorkspace` tests. I can take this one off.

Comment (mfshao) [reply bmh_admin_portal_ui/src/views/RequestWorkspace.test.js]: do we need this in other tests? are we firing requests to this URL at all for any of the tests?

**Review by mfshao — APPROVED** [review]

**Pull request merged**